src/Compilers/CSharp/Portable/BoundTree/BoundImplicitIndexerAccess.cs

```diff
@@ -29,5 +29,11 @@ internal BoundExpression GetReceiver()
             Debug.Assert(receiver is not null);
             return receiver;
         }
+
+        private partial void Validate()
+        {
+            Debug.Assert(LengthOrCountAccess is BoundPropertyAccess or BoundLocal or BoundBadExpression);
+            Debug.Assert(IndexerOrSliceAccess is BoundIndexerAccess or BoundCall or BoundArrayAccess or BoundBadExpression);
```

Comment:
> or BoundBadExpression [](http://example.com/codeflow?start=100&length=22)

The addition of a BoundBadExpression for IndexerOrSliceAccess feels unexpected. We don't call CheckValue on it and we don't handle BoundBadExpression in various helpers. For example in GetReceiver above.